src/compute/src/render/join/delta_join.rs

```diff
@@ -604,10 +606,11 @@ where
                             while let Some(key) = cursor.get_key(batch) {
                                 while let Some(val) = cursor.get_val(batch) {
                                     cursor.map_times(batch, |time, diff| {
+                                        let time = time.into_owned();
                                         // note: only the delta path for the first relation will see
                                         // updates at start-up time
                                         if source_relation == 0
-                                            || !inner_as_of.elements().contains(time)
+                                            || !inner_as_of.elements().contains(&time)
```

Comment:
Nit: can we do this comparison based on the timegat version? It seems like it means the difference between cloning all snapshot times only to discard them, vs not.